### ISSUE TITLE
DeploymentOptions setMaxWorkerExecuteTime

### ISSUE THREAD

**Issue by b-liberman** [vert-x3/issues#203]
I do not see that the method DeploymentOptions.setMaxWorkerExecuteTime has any effect.

Looking into the code I do not see that the method getMaxWorkerExecuteTime is invoked anywhere and I do not see any other way how the attribute maxWorkerExecuteTime can be used.

In contrast, the attribute maxWorkerExecuteTime in the VertxOptions class is used properly.

**Comment by vietj** [comment]
hi, you need to use a named thread pool to make it work

**Comment by vietj** [comment]
with workerPoolName

**Comment by b-liberman** [comment]
hi, thanks a lot. Could you perhaps provide an example? I did set the worker pool name in the deployment options but it also did not help and the max worker execute time set on the deployment options was not used. Here is my code where I set the max execute time on VertxOptions as it did not work when applied to the DeploymentOptions:

Vertx vertx = Vertx.vertx(new VertxOptions().setWorkerPoolSize(10).setMaxWorkerExecuteTime(180L \* 1000 \* 1000000));
        DeploymentOptions deploymentOptions = new DeploymentOptions().setWorkerPoolName("generic-request-processor-pool")
            .setWorker(true).setMultiThreaded(true);
        vertx.deployVerticle(new GenericRequestProcessor(), deploymentOptions);

**Comment by vietj** [comment]
it should be:

```
DeploymentOptions deploymentOptions = new DeploymentOptions().setWorkerPoolName("generic-request-processor-pool").setMaxWorkerExecuteTime(limit);
```

**Comment by b-liberman** [comment]
I tried it - this did not work for me. Again, the source code does not look like the property maxWorkerExecuteTime can be used.

**Comment by PennyYip** [comment]
What @b-liberman said is true (As of 3.4.2), only the maxWorkerExecuteTime being used is in VertxOption. 

The maxWorkerExecuteTime in DeploymentOption is not used in any context except for the DeploymentOptionsConverter, which used to convert Json configurations.

**Comment by b-liberman** [comment]
as the issue is verified by others, can it be re-opened, please?

**Comment by scoolin** [comment]
@b-liberman is right.
in the DeploymentManager, line 468 (version 3.5.0):

`WorkerExecutorImpl workerExec = poolName != null ? vertx.createSharedWorkerExecutor(poolName, options.getWorkerPoolSize()) : null;`

Deploy new verticle dose not use the 3rd arg of createSharedWorkerExecutor(), so the VertxOptions.getMaxWorkerExecuteTime() will take effect.

**Comment by tsegismont** [comment]
Reopening. This test fails in master:

```java
  @Test
  public void testMaxExecuteWorkerTime() throws Exception {
    String poolName = TestUtils.randomAlphaString(10);
    long maxWorkerExecuteTime = NANOSECONDS.convert(3, SECONDS);
    DeploymentOptions deploymentOptions = new DeploymentOptions()
      .setWorkerPoolName(poolName)
      .setMaxWorkerExecuteTime(maxWorkerExecuteTime);
    vertx.deployVerticle(new AbstractVerticle() {
      @Override
      public void start(Future<Void> startFuture) throws Exception {
        vertx.executeBlocking(fut -> {
          try {
            SECONDS.sleep(5);
            fut.complete();
          } catch (InterruptedException e) {
            fut.fail(e);
          }
        }, startFuture);
      }
    }, deploymentOptions, onSuccess(did -> {
      testComplete();
    }));
    await();
    String timeLimitMsg = "time limit is " + MILLISECONDS.convert(maxWorkerExecuteTime, NANOSECONDS);
    assertTrue(Files.lines(Paths.get(System.getProperty("java.io.tmpdir"), "vertx.log"))
      .anyMatch(line -> {
        if (line.contains("has been blocked for")
          && line.contains(timeLimitMsg)
          && line.contains(poolName)) {
          return true;
        }
        return false;
      }));
  }
```

**Comment by tsegismont** [comment]
As a workaround, create a `SharedWorkerExecutor` before deploying the verticle with `io.vertx.core.Vertx#createSharedWorkerExecutor(java.lang.String, int, long)`

**Comment by tsegismont** [comment]
Will be fixed in 3.5.1
